src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.common.collect.ImmutableList;
+import htsjdk.variant.vcf.VCFHeader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Service to create BigQuery Schema from VCFHeader 
+ */
+public class SchemaGeneratorImpl implements SchemaGenerator {
+  
+  public TableSchema getSchema(VCFHeader vcfHeader){
+    ImmutableList<TableFieldSchema> schemaFields = getFields(vcfHeader);
+    TableSchema schema = new TableSchema()
+        .setFields(schemaFields);
+    return schema;
+  }
+
+  public ImmutableList<TableFieldSchema> getFields(VCFHeader vcfHeader){
+    ImmutableList.Builder<TableFieldSchema> fields = new ImmutableList.Builder<TableFieldSchema>();
+    Integer[] fieldIndices = getFieldIndices(SchemaUtils.constantFieldIndexToNameMap);
```

Comment:
If we need to order the indices, why not using a ordered map(TreeMap) to store the keys in order

Reply:
Awesome idea! I refactored this part to use a TreeMap.